src/devtools/client/inspector/rules/components/RulesApp.js

```diff
@@ -183,7 +183,7 @@ class RulesApp extends PureComponent {
     const pseudoElementRules = [];
     const styleRules = [];
 
-    for (const rule of rules) {
+    for (const rule of rules || []) {
```

Comment:
could we ensure the rules are always an array?

Reply:
No, that's the point of this PR: we need to distinguish between the situation where `getAppliedRules` succeeds and returns 0 rules versus the situation where it fails. But in this place we don't need that distinction.

Reply:
got it, that makes more sense. 

Another way to do this would be to introduce a `groupRules` function that returns `inherited`, `pseudoElements`, and `styleRules`  and factor out the branching into `renderPane`:

that would look like this:

```js
renderPane() {
  if (!rules) { 
     return <div className="devtools-sidepanel-no-result" >No Rules</div>
  }

  if (rules.length == 0) {
   return <div className="devtools-sidepanel-no-result" >Rules not available</div>
  }

  const { styled, pseudoElements, inherited } = this.groupRules(rules)
  return <>
   {this.renderPseudoElementRules(pseudoElementRules)}
   {this.renderStyleRules(styleRules)}
   {this.renderInheritedRules(inheritedRules)}
  </>
}

render() {
  return <div id="sidebar-panel-ruleview" className="theme-sidebar inspector-tabpanel ">
    {this.renderPane()}
  </div>
}
```

This is cosmetic, but i think it is nice to move branching into functions with their own returns. 

Also, i'd like to start using JSX where we can in the elements panel... It can be gradual, but it is much nicer to read.

Also, I'd like to get rid of getStr and l10n globally. The properties files are big and are making our build process much more complicated.